tests/test_grid3d/test_grid_properties.py

```diff
@@ -81,6 +88,42 @@ def test_gridproperties_iter():
     assert count == 4
 
 
+@given(gridproperties())
+def test_gridproperties_copy(grid_properties):
+    grid_properties_copy = grid_properties.copy()
+
+    assert grid_properties_copy.dates == grid_properties.dates
+    assert grid_properties_copy.names == grid_properties.names
+
+    assume(grid_properties.props)
+    assume(grid_properties.props[0].date != 19990101)
+    grid_properties_copy.props[0].date = 19990101
+    assert grid_properties.props[0].date != 19990101
+
+
+@given(gridproperties_elements(), gridproperties_elements())
+def test_consistency_check(gridproperty1, gridproperty2):
+    assume(gridproperty1.dimensions != gridproperty2.dimensions)
+    with pytest.raises(ValueError, match="Mismatching dimensions"):
+        GridProperties(props=[gridproperty1, gridproperty2])
+
+
+@given(gridproperties_elements())
+def test_gridproperties_from_roff(grid_property):
+    buff = io.BytesIO()
+    grid_property.to_file(buff, fformat="roff")
+    buff.seek(0)
+    xtgeo.gridproperties_from_file(buff, fformat="roff", names=[grid_property.name])
+
```

Comment:
Perhaps verify that the object read == original object?

Reply:
My bad, forgot to add an assert here. We cannot have a direct `==` on gridproperties here as the underlying object GridProperty cannot be directly compared (something we would like to fix, but we are far off from that). We can, however, assert that the names are the same.